src/Pos/Explorer/Web/Server.hs

```diff
@@ -126,62 +147,124 @@ getAddressSummary cAddr = cAddrToAddr cAddr >>= \addr -> case addr of
     PubKeyAddress sid _ -> do
         balance <- fromMaybe (mkCoin 0) <$> GS.getFtsStake sid
         -- TODO: add number of coins when it's implemented
-        txs <- allTxs
-        let transactions = map (toTxDetailed addr) $
-                           filter (any (\txOut -> txOutAddress txOut == addr) .
-                               _txOutputs . tiTx) txs
+        -- TODO: retrieve transactions from something like an index
+        let transactions = []
         return $ CAddressSummary cAddr 0 balance transactions
     _ -> throwM $
          Internal "Non-P2PKH addresses are not supported in Explorer yet"
 
+getTxSummary :: ExplorerMode m => CTxId -> m CTxSummary
+getTxSummary cTxId = do
+    -- There are two places whence we can fetch a transaction: MemPool and DB.
+    txId <- cTxIdToTxId cTxId
+    txExtraMaybe <- GS.getTxExtra txId
+
+    when (isNothing txExtraMaybe) $
+        throwM $ Internal "Transaction not found"
+
+    let txExtra = fromJust txExtraMaybe
```

Comment:
hm, this would work fine now.... but if somebody by mistake removes L163 then we have run time bug. Instead of this I would prefer either pattern match on `case` or, even better, do `teExtra <- maybe (throwM ...) pure teExtraMaybe`